{{cookiecutter.project_dash_slug}}/Dockerfile

```diff
@@ -0,0 +1,42 @@
+# Base Image
+FROM python:3.6.6-alpine3.7
+
+# Best practice: if you build contains several layers you should order them
+# from the less frequently changed to the more frequently changed (caching)
+
+# Install tools required for the project
+# Best practice:
+# 1. do not install unnecessary packages, keep it small.
+# 2. keep an alphabetic order to avoid duplicates
+RUN apk add --no-cache --update \
+    build-base \
+    git \
+    libffi-dev \
+    linux-headers \
+    pcre-dev \
+    postgresql-client \
+    python3-dev
+
+# Run as non root user (best practice)
+ENV USERNAME=syapse
+ENV PROJECT_NAME={{cookiecutter.project_dash_slug}}
+WORKDIR /home/$USERNAME/$PROJECT_NAME
+COPY . .
+RUN addgroup -g 303 -S $USERNAME \
+    && adduser -u 303 -S -G $USERNAME -h /home/$USERNAME $USERNAME
```

Comment:
You can simplify this block a lot: when you create a user, it'll automatically create a group and home directory for it:

```bash
# Creates a user, its home directory, and a group with the same name.
adduser $USERNAME

# Even simpler, if you took my other suggestion:
adduser user
```

Reply:
@eabruzzese  I don't like the idea of having a user names user ? can we just call it something else ?

Reply:
@gianlucabiccari Just keep in mind that we don't many users -- it's just the super user and the not-super user... What would you like to call it?

Reply:
syapse ?